### ISSUE TITLE
Is there anyway to turn off progress bar when do evaluation?

### ISSUE THREAD

**Issue by darcula1993** [open-mmlab/mmpose#1314]

**Comment by liqikai9** [comment]
You can comment these lines:

https://github.com/open-mmlab/mmpose/blob/4853b4bcd1238ef559c0f341ed2402c5a3605316/mmpose/apis/test.py#L38-L39

if you single GPU. 

Or you can comment these lines:
https://github.com/open-mmlab/mmpose/blob/4853b4bcd1238ef559c0f341ed2402c5a3605316/mmpose/apis/test.py#L76-L77

if you use multiple GPUs.

BTW, is there any inconvenience you have using the progress bar tool?

**Comment by darcula1993** [comment]
@liqikai9 The training server I use do not have interactive terminal, so we have to save the print log to disk to check. 
And here is the log,the progress bar just print each line for one step even with multiple gpus.
Should I update mmpose? the curent version is 0.21.0.
 
`2022-04-14 03:26:37,778 - mmpose - INFO - Epoch(val) [120][407]	AP: 0.1732, AP .5: 0.4812, AP .75: 0.0907, AP (M): 0.1173, AP (L): 0.1541, AR: 0.3098, AR .5: 0.6588, AR .75: 0.2571, AR (M): 0.2989, AR (L): 0.3255
2022-04-14 03:32:14,867 - mmpose - INFO - Epoch [121][200/293]	lr: 5.000e-06, eta: 8:31:11, time: 1.685, data_time: 0.887, memory: 840, reg_loss: 5.2262, acc_pose: 0.4311, loss: 5.2262
2022-04-14 03:40:20,490 - mmpose - INFO - Epoch [122][200/293]	lr: 5.000e-06, eta: 8:25:27, time: 1.716, data_time: 0.242, memory: 840, reg_loss: 5.2321, acc_pose: 0.4281, loss: 5.2321
2022-04-14 03:48:21,497 - mmpose - INFO - Epoch [123][200/293]	lr: 5.000e-06, eta: 8:19:44, time: 1.717, data_time: 0.264, memory: 840, reg_loss: 5.2341, acc_pose: 0.4281, loss: 5.2341
2022-04-14 03:56:21,234 - mmpose - INFO - Epoch [124][200/293]	lr: 5.000e-06, eta: 8:13:59, time: 1.707, data_time: 0.513, memory: 840, reg_loss: 5.2346, acc_pose: 0.4327, loss: 5.2346
2022-04-14 04:04:25,062 - mmpose - INFO - Epoch [125][200/293]	lr: 5.000e-06, eta: 8:08:16, time: 1.719, data_time: 0.234, memory: 840, reg_loss: 5.2325, acc_pose: 0.4336, loss: 5.2325
2022-04-14 04:12:30,256 - mmpose - INFO - Epoch [126][200/293]	lr: 5.000e-06, eta: 8:02:32, time: 1.715, data_time: 0.305, memory: 840, reg_loss: 5.2358, acc_pose: 0.4308, loss: 5.2358
2022-04-14 04:20:32,292 - mmpose - INFO - Epoch [127][200/293]	lr: 5.000e-06, eta: 7:56:48, time: 1.713, data_time: 0.220, memory: 840, reg_loss: 5.2427, acc_pose: 0.4324, loss: 5.2427
2022-04-14 04:28:35,095 - mmpose - INFO - Epoch [128][200/293]	lr: 5.000e-06, eta: 7:51:04, time: 1.713, data_time: 0.233, memory: 840, reg_loss: 5.2389, acc_pose: 0.4335, loss: 5.2389
2022-04-14 04:36:37,158 - mmpose - INFO - Epoch [129][200/293]	lr: 5.000e-06, eta: 7:45:20, time: 1.710, data_time: 0.257, memory: 840, reg_loss: 5.2345, acc_pose: 0.4325, loss: 5.2345
2022-04-14 04:44:40,907 - mmpose - INFO - Epoch [130][200/293]	lr: 5.000e-06, eta: 7:39:38, time: 1.727, data_time: 0.227, memory: 840, reg_loss: 5.2377, acc_pose: 0.4304, loss: 5.2377
2022-04-14 04:47:01,280 - mmpose - INFO - Saving checkpoint at 130 epochs
[                                                  ] 0/104125, elapsed: 0s, ETA:
[                           ] 1/104125, 0.0 task/s, elapsed: 29s, ETA: 3049871s
[                           ] 2/104125, 0.1 task/s, elapsed: 29s, ETA: 1524937s
[                           ] 3/104125, 0.1 task/s, elapsed: 29s, ETA: 1016617s
[                            ] 4/104125, 0.1 task/s, elapsed: 29s, ETA: 762456s
[                            ] 5/104125, 0.2 task/s, elapsed: 29s, ETA: 609960s
[                            ] 6/104125, 0.2 task/s, elapsed: 29s, ETA: 508296s
[                            ] 7/104125, 0.2 task/s, elapsed: 29s, ETA: 435678s
[                            ] 8/104125, 0.3 task/s, elapsed: 29s, ETA: 381215s
[                            ] 9/104125, 0.3 task/s, elapsed: 29s, ETA: 338855s
[                           ] 10/104125, 0.3 task/s, elapsed: 29s, ETA: 304967s
[                           ] 11/104125, 0.4 task/s, elapsed: 29s, ETA: 277241s
[                           ] 12/104125, 0.4 task/s, elapsed: 29s, ETA: 254135s
[                           ] 13/104125, 0.4 task/s, elapsed: 29s, ETA: 234584s
[                           ] 14/104125, 0.5 task/s, elapsed: 29s, ETA: 217826s
[                           ] 15/104125, 0.5 task/s, elapsed: 29s, ETA: 203303s
`

**Comment by liqikai9** [comment]
I would like to reproduce your problem, but I need more information. So could you please offer this information?

- What command or script did you run?

- What config did you run?

- Did you make any modifications on the code or config? 

**About the Environment**
1. Please run `PYTHONPATH=${PWD}:$PYTHONPATH python mmpose/utils/collect_env.py` to collect necessary environment information and paste it here.

**Comment by darcula1993** [comment]
comment these lines works,

```
model.eval()
results = []
dataset = data_loader.dataset
rank, world_size = get_dist_info()
# if rank == 0:
    # prog_bar = mmcv.ProgressBar(len(dataset))
for data in data_loader:
    with torch.no_grad():
        result = model(return_loss=False, **data)
    results.append(result)

    # if rank == 0:
        # # use the first key as main key to calculate the batch size
        # batch_size = len(next(iter(data.values())))
        # for _ in range(batch_size * world_size):
            # prog_bar.update()
```
seems that `if rank == 0` condition do not work

**Comment by darcula1993** [comment]
It is a local server with 8 2080ti and inspur training platform. CentOS7, pytorch 1.7.0 with cuda 11.0,Driver Version: 450.102.04

**Comment by darcula1993** [comment]
```
[1] 2022-04-14 09:10:34,807 - mmpose - INFO - Environment info:
[2] ------------------------------------------------------------
[3] sys.platform: linux
[4] Python: 3.6.9 (default, Oct  8 2020, 12:12:24) [GCC 8.4.0]
[5] CUDA available: True
[6] GPU 0,1,2,3,4,5,6: GeForce RTX 2080 Ti
[7] CUDA_HOME: /usr/local/cuda
[8] NVCC: Build cuda_11.0_bu.TC445_37.28845127_0
[9] GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
[10] PyTorch: 1.7.0+cu110
[11] PyTorch compiling details: PyTorch built with:
[12]   - GCC 7.3
[13]   - C++ Version: 201402
[14]   - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
[15]   - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
[16]   - OpenMP 201511 (a.k.a. OpenMP 4.5)
[17]   - NNPACK is enabled
[18]   - CPU capability usage: AVX2
[19]   - CUDA Runtime 11.0
[20]   - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80
[21]   - CuDNN 8.0.4
[22]   - Magma 2.5.2
[23]   - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 
[24] 
[25] TorchVision: 0.8.1+cu110
[26] OpenCV: 4.5.4
[27] MMCV: 1.4.0
[28] MMCV Compiler: GCC 7.3
[29] MMCV CUDA Compiler: 11.0
[30] MMPose: 0.21.0+228747c
```
